### PR TITLE
add iziswap lp

### PR DESCRIPTION
**Pull request by buchaoqun**
**NOTE**
add iziswap lp

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at undefined: 
        
 
  \n Please revert changes to package.json / package-lock.json

**Comment by llamatester** [comment]
Error while running adapter at undefined: 
        
 
  \n Please revert changes to package.json / package-lock.json

**Comment by llamatester** [comment]
The adapter at projects/solv-protocol-funds exports TVL: 
        
 
 
    arbitrum                  61.64 M
    mantle                    252.71 k
    arbitrum-borrowed         7.27 M
    mantle-borrowed           121.26 k
    borrowed                  7.39 M
    
    total                    61.89 M

File: projects/solv-protocol-funds/index.js [hunk 1]
```diff
@@ -103,9 +104,84 @@ async function mantleTvl(ts, _, _1, { api }) {
     api.add(klp["address"], amount)
   })
 
+  await iziswap(ts, _, _1, { api });
+
   return api.getBalances()
 }
 
+async function iziswap(ts, _, _1, { api }) {
```
Comment (g1nt0ki) [inline projects/solv-protocol-funds/index.js:112]: I should create a helper (like the one for uniswap v3 NFTs) at some point so people dont have to create this each time.

**Comment by llamatester** [comment]
The adapter at projects/solv-protocol-funds exports TVL: 
        
 
 
    arbitrum                  61.42 M
    mantle                    252.46 k
    arbitrum-borrowed         7.40 M
    mantle-borrowed           120.93 k
    borrowed                  7.53 M
    
    total                    61.67 M

File: projects/solv-protocol-funds/iziswap.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+function _getAmountY(
+    liquidity,
+    sqrtPriceL,
+    sqrtPriceR,
+    sqrtRate,
+    upper,
+) {
+    const numerator = sqrtPriceR - sqrtPriceL;
+    const denominator = sqrtRate - 1;
+    if (!upper) {
```
[the remark this answers is not in the document]
Comment (g1nt0ki) [reply projects/solv-protocol-funds/iziswap.js]: @buchaoqun thanks for the PR, I ma merging your changes, but can you please answer these questions when you get some time?

[the remark this answers is not in the document]
Comment (buchaoqun) [reply projects/solv-protocol-funds/iziswap.js]: Sorry to make you wait for a long time, this processing method is exactly the same as my official sdk method in iziSwap. 
Link: https://github.com/izumiFinance/iZiSwap-sdk/blob/main/src/liquidityManager/library/amountMath.ts

File: projects/solv-protocol-funds/index.js (hunk 1 above)
Comment (buchaoqun) [reply projects/solv-protocol-funds/index.js]: Yes, I think this is best too.

**Pull request merged**